logstash-core/lib/logstash/elasticsearch_client.rb

```diff
@@ -39,8 +39,11 @@ def initialize(settings, logger)
       @client_args[:ssl] = ssl_options
 
       username = @settings["var.elasticsearch.username"]
-      password = @settings["var.elasticsearch.password"]
       if username
+        password = @settings["var.elasticsearch.password"]
+        if password.is_a?(LogStash::Util::Password)
```

Comment:
when `@settings` gets to elasticsearch_client its just a hash. Anything could have happened to it - its mutable. The es_client is not exclusively for module use.